libs/sdk-common/src/input_parser.rs

```diff
@@ -197,6 +219,58 @@ pub async fn parse(
     Err(anyhow!("Unrecognized input type"))
 }
 
+async fn bip353_parse(
+    input: &str,
+    dns_resolver: &AsyncResolver<GenericConnector<TokioRuntimeProvider>>,
+) -> Option<String> {
+    if let Some((local_part, domain)) = input.split_once('@') {
+        let dns_name = format!("{}.{}.{}", local_part, USER_BITCOIN_PAYMENT_PREFIX, domain);
+
+        // Query for TXT records of a domain
+        let txt_data = match dns_resolver.txt_lookup(dns_name).await {
+            Ok(records) => records
+                .iter()
+                .flat_map(|record| record.to_string().into_bytes())
+                .collect::<Vec<u8>>(),
+            Err(e) => {
+                debug!("No BIP353 TXT records found: {}", e);
+                return None;
+            }
+        };
+
+        // Decode TXT data
+        match String::from_utf8(txt_data) {
+            Ok(decoded) => {
+                if !decoded.to_lowercase().starts_with(BIP353_PREFIX) {
+                    error!(
+                        "Invalid decoded TXT data (doesn't begin with: {})",
+                        BIP353_PREFIX
+                    );
+
+                    return None;
+                }
+
+                if let Some((_, bolt12_address)) =
```

Comment:
I would suggest to return the full BIP21 string here. And have the rest of the input parser handle the resulting string. For example, I found this string in the wild: `bitcoin:?lno=lno1zrxq8pjw7qjlm68mtp7e3yvxee4y5xrgjhhyf2fxhlphpckrvevh50u0qffqdtuhyd77jpju7prz0qvkcm0xfzfpz844z9dwkaexuapm3e66vqsrxkhw43wz8wfa349nyl2wcsdp2u62ykz4lzuke37p75f4x286v6eqqva25lyzapjs7h424f7v65hv3k539558aqdztn24tjtkyqrrnr3f4vxwxhz0hhrkmzt4mz228sgs3tjl2vzhq0y3s4g8c65nvl35gyy2r833d6qaf64e2qwkey8zkm2q4yln60xgkqqsz8u6nrn984hru24yunnk0p093c&sp=sp1qqvwqe7yvuk9hwy6keczg2h0xajp6zeqzamzpcpfm0nrpgjr42yyq2q6hzkuldhpj5frduwjlzndf26rk9atzt49qwufun7drfw3c8yxypvk57ux5`

Reply:
The problem I see here is that this doesn't look like a valid bip21 as it misses the address part (before the query params).

Reply:
@roeierez The address field is not mandatory and most parsers support it, so it should be alright

Reply:
@roeierez The mandatory address is a disputed property. I think it shouldn't be necessary to enforce that, especially as a client.

Reply:
I agree, just stated that our current bip21 parser (the crate we use) doesn't seem to support that.

Reply:
I did another test and confirm that the current bip21 parser returns that the provided bip21 is not a valid bip21uri.

I checked if the latest version of the crate still require the address and the answer is yes unfortunatly. Do you have some advice to proceed?

Reply:
I would go for the simple (and practical) approach now .Means parse the uri ourselves and the query params.
If contains offer return the offer string.
If contains lnurl return the lnurl string.

Reply:
I extracted the parsing functions to be easy reusable when we want to move the logic. I added the feature = "liquid" for the bolt12 part.